include/CLI/Set.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+// Distributed under the 3-Clause BSD License.  See accompanying
+// file LICENSE or https://github.com/CLIUtils/CLI11 for details.
+
+#include <initializer_list>
+#include <memory>
+#include <string>
+#include <vector>
+
+// CLI Library includes
+#include "CLI/StringTools.hpp"
+
+namespace CLI {
+
+class Set {
```

Comment:
I wonder about the possibility of making set a class template, then you could have sets of <int>, or other things as long as they had a lexical cast or `>>` operator.  
```
Option *add_cli_set(std::string option_name,
                        T &member,   ///< The selected member of the set
                        Set<T> options, ///< The set of possibilities
                        std::string description = "") {
```

Reply:
Along with a specialization for string that has the extra ignore_case and ignore_underscore functions

Reply:
I thought about that, I just left it off for simplicity for now. I'll probably add it as part of making it work with enum outputs. However, since this is for "possible choices", I think requiring strings would not be too bad if it can't be done.